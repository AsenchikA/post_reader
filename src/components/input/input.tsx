import React, { FC } from 'react';
import classNames from 'classnames';
import styles from './input.css';

interface IInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  id: string;
  label?: string;
  className?: string;
}

export const Input: FC<IInputProps> = (props) => {
  const { className, label, id, ...restProps } = props;
  return (
    <div className={classNames(styles.container, className)}>
      {label && <label htmlFor={id}>{label}</label>}
      <input id={id} className={styles.input} {...restProps} />
    </div>
  );
};
