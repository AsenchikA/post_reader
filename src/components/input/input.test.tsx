import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from './input';

it('Input renders correcly', () => {
  const props = {
    id: 'test-id',
    label: 'label',
    className: 'test-class',
  };

  const tree = renderer.create(<Input {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
