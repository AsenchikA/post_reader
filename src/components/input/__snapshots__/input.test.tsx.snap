// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correcly 1`] = `
<div
  className="container test-class"
>
  <label
    htmlFor="test-id"
  >
    label
  </label>
  <input
    className="input"
    id="test-id"
  />
</div>
`;
